tests/unit/dogstatsd/test_statsd.py

```diff
@@ -767,6 +767,59 @@ def test_entity_tag_and_tags_from_environment_and_constant(self):
         tags = "country:canada,red,country:china,age:45,blue,dd.internal.entity_id:04652bb7-19b7-11e9-9cc6-42010a9c016d"
         assert_equal_telemetry('gt:123.4|g|#'+tags, statsd.socket.recv(), telemetry=telemetry_metrics(tags=tags))
 
+    def test_dogstatsd_initialization_with_dd_env_service_version(self):
+        """
+        Dogstatsd will automatically use DD_ENV, DD_SERVICE, and DD_VERSION
+        to set the {env, service, version} tags for all metrics emitted.
+
+        This test covers several possible cases of tag configuration.
+        Tags configured from any source are added in an append-only fashion.
+        """
+        cases = [
+            ('prod', 'dog', 'abc123', '', [], ['env:prod', 'service:dog', 'version:abc123']),
+            ('prod', 'dog', 'abc123', 'env:prod,type:app', [], ['env:prod', 'type:app', 'env:prod', 'service:dog', 'version:abc123']),
+            ('prod', 'dog', 'abc123', 'env:prod2,type:app', [], ['env:prod2', 'type:app', 'env:prod', 'service:dog', 'version:abc123']),
+            ('prod', 'dog', 'abc123', '', ['env:prod', 'type:app'], ['env:prod', 'type:app', 'env:prod', 'service:dog', 'version:abc123']),
+            ('prod', 'dog', 'abc123', '', ['env:prod2', 'type:app'], ['env:prod2', 'type:app', 'env:prod', 'service:dog', 'version:abc123']),
+            ('prod', 'dog', 'abc123', 'env:prod3,custom_tag:cat', ['env:prod2', 'type:app'], ['env:prod2', 'type:app', 'env:prod3', 'custom_tag:cat', 'env:prod', 'service:dog', 'version:abc123']),
+        ]
+        for c in cases:
+            dd_env, dd_service, dd_version, datadog_tags, constant_tags, global_tags = c
+            with preserve_environment_variable('DATADOG_TAGS'):
+                os.environ['DATADOG_TAGS'] = datadog_tags
+                with preserve_environment_variable('DD_ENV'):
+                    os.environ['DD_ENV'] = dd_env
+                    with preserve_environment_variable('DD_SERVICE'):
+                        os.environ['DD_SERVICE'] = dd_service
+                        with preserve_environment_variable('DD_VERSION'):
+                            os.environ['DD_VERSION'] = dd_version
+                            statsd = DogStatsd(constant_tags=constant_tags, telemetry_min_flush_interval=0)
+                            statsd.socket = FakeSocket()
```

Comment:
```suggestion
            with (
                preserve_environment_variable('DATADOG_TAGS'), 
                preserve_environment_variable('DD_ENV'),
                preserve_environment_variable('DD_SERVICE'), 
                preserve_environment_variable('DD_VERSION'),
            ):
                os.environ['DATADOG_TAGS'] = datadog_tags
                os.environ['DD_ENV'] = dd_env
                os.environ['DD_SERVICE'] = dd_service
                os.environ['DD_VERSION'] = dd_version
                statsd = DogStatsd(constant_tags=constant_tags, telemetry_min_flush_interval=0)
                statsd.socket = FakeSocket()
```

Reply:
Great suggestion! I always forget about this formatting.